docs/pages/_app.js

```diff
@@ -1,19 +1,114 @@
-import React from 'react'
+import React from 'react';
 import * as Sentry from '@sentry/browser';
-import App, { Container } from 'next/app'
+import App, { Container } from 'next/app';
 
-Sentry.init({dsn: "https://67e35a01698649d5aa33aaab61777851@sentry.io/1526800"});
+Sentry.init({
+  dsn: 'https://67e35a01698649d5aa33aaab61777851@sentry.io/1526800',
+  beforeSend(event) {
+    let message = getMessage(event);
 
-class MyApp extends App {
+    // If we don't know about this particular type of event then just pass it along
+    if (!message) {
+      return event;
+    }
+
+    // Discard any errors that we know we do not care about
+    if (ERRORS_TO_DISCARD.includes(message)) {
+      return null;
+    }
+
+    // Only attempt to check against cached reported messages if we have localStorage
+    if (window.localStorage) {
+      // Clear the saved error messages every day
+      maybeResetReportedErrorsCache();
+
+      // Bail out if we have reported the error already
+      if (userHasReportedErrorMessage(message)) {
+        return null;
+      }
+
+      saveReportedErrorMessage(message);
+    }
+
+    return event;
+  },
+});
+
+export default class MyApp extends App {
   render() {
-    const { Component, pageProps } = this.props
+    const { Component, pageProps } = this.props;
 
     return (
       <Container>
         <Component {...pageProps} />
       </Container>
-    )
+    );
   }
 }
 
-export default MyApp
\ No newline at end of file
+/*
+ * Error logging filtering - prevent users from submitting errors we do not care about,
+ * eg: specific error messages that are caused by extensions or other scripts
+ * out of our control, or the same error being reported many times.
+ */
+
+// These exact error messages may be different depending on the browser!
+const ERRORS_TO_DISCARD = [
+  // This error only appears in Safari
+  "undefined is not an object (evaluating 'window.__pad.performLoop')",
+];
+
+const REPORTED_ERRORS_KEY = 'sentry:reportedErrors';
+const TIMESTAMP_KEY = 'sentry:errorReportingInit';
+const ONE_DAY_MS = 60 * 60 * 24 * 1000;
+
+// Extract a stable event error message out of the Sentry event object
+function getMessage(event) {
+  if (event.message) {
+    return event.message;
+  }
+
+  if (event.exception && event.exception.values) {
```

Comment:
Do we need to handle the case for single exceptions? I read in the sentry docs that some circumstances result in exceptions w/o a `values` array, just a single `value` ->https://docs.sentry.io/development/sdk-dev/event-payloads/exception/#examples

Reply:
strange! in my tests even single exceptions resulted in an array of values. i guess let's see how it goes and if it doesn't seem to be working then we can adjust, or if you want to open a pr to fix it that'd work too